lib/twirp/client.rb

```diff
@@ -120,6 +120,10 @@ def make_http_request(conn, service_full_name, rpc_method, content_type, req_opt
               r.headers[k] = v
             end
           end
+
+          if req_opts && req_opts[:timeout] && req_opts[:timeout].is_a?(Integer) && req_opts[:timeout] > 0
```

Comment:
I made a choice to ignore non positive Integer values because Faraday doesn't actually check this and fails rather hard with a `NoMethod` error if you don't pass in a number. If ignoring invalid values doesn't sound good, we could instead raise a `ArgumentError` here and make the request fail.